### PR TITLE
Launch process for metrics calculation only once per project

### PR DESCRIPTION
**Pull request by vilchik-elena**
Fixes SonarSource/SonarTS#116

### CODE REVIEW

File: sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ExternalTypescriptSensorTest.java [hunk 1]
```diff
@@ -142,16 +140,15 @@ private String resourceScript(String script) throws URISyntaxException {
...
     thrown.expectMessage("Failed to run external process `non_existent_command arg1`");
 
-    // fails even without single file in project as command run once per command
-    // as there is no files, tsMetrics is not executed
     createSensor(testBundle).execute(createSensorContext());
   }
 
+  @Ignore
   @Test
   public void should_log_when_failed_ts_metrics_process() throws Exception {
```
Comment (SonarTech) [inline sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ExternalTypescriptSensorTest.java]: MAJOR Fix or remove this skipped unit test [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1607)

File: sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 2]
```diff
@@ -104,11 +104,25 @@ private void runMetrics(SensorContext sensorContext, ExecutableBundle executable
...
+        saveCpd(sensorContext, tsMetricsPerFileResponse.cpdTokens, inputFile);
+      } else {
+        LOG.error("During metric calculation failed to find input file for path '" + tsMetricsPerFileResponse.filepath + "'");
+      }
+    }
   }
 
 
-  private void runMetricsForFile(SensorContext sensorContext, ExecutableBundle executableBundle, InputFile file, InputFileContentExtractor contentExtractor) {
+  private TsMetricsPerFileResponse[] runMetricsProcess(SensorContext sensorContext, ExecutableBundle executableBundle, Iterable<InputFile> inputFiles, InputFileContentExtractor contentExtractor) {
```
Comment (SonarTech) [inline sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: MAJOR Make "runMetricsProcess" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)
MAJOR Remove this unused method parameter "sensorContext". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)
MINOR Split this 196 characters long line (which is greater than 180 authorized). [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ExternalTypescriptSensorTest.java [hunk 3]
```diff
@@ -163,28 +158,26 @@ public void should_log_when_failed_ts_metrics_process() throws Exception {
     DefaultInputFile testInputFile = createTestInputFile(sensorContext);
```
Comment (saberduck) [inline sonar-typescript-plugin/src/test/java/org/sonar/plugin/typescript/ExternalTypescriptSensorTest.java]: @vilchik-elena seems that input file is not needed anymore?

File: sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java [hunk 4]
```diff
@@ -109,13 +107,25 @@ private void runMetrics(SensorContext sensorContext, ExecutableBundle executable
...
+      } else {
+        LOG.error("During metric calculation failed to find input file for path '" + tsMetricsPerFileResponse.filepath + "'");
+      }
+    }
   }
 
 
-  private void runMetricsForFile(SensorContext sensorContext, ExecutableBundle executableBundle, InputFile file, InputFileContentExtractor contentExtractor) {
+  private static TsMetricsPerFileResponse[] runMetricsProcess(ExecutableBundle executableBundle, Iterable<InputFile> inputFiles) {
+    List<String> filepaths = new ArrayList<>();
```
Comment (saberduck) [inline sonar-typescript-plugin/src/main/java/org/sonar/plugin/typescript/ExternalTypescriptSensor.java]: we can use guava's `Iterables`  instead

**Pull request merged**